Automattic/woocommerce-payments
# Prevent proceeding to WooPay's checkout on previews

Fixes #4985

#### Changes proposed in this Pull Request

* Check for the URL parameter used in the customize.php iframe.
* Check for the is_preview() value.
* Prevent proceeding to WooPay checkout when any of them are true.

For the customize.php preview, I evaluated using backend resources for doing the check, like the value of `$pagenow` and hooking to [customize_controls_init](https://developer.wordpress.org/reference/hooks/customize_controls_init/). But this approach wouldn't work because their values change when navigating withing the preview page.

#### Testing instructions

1. Log in with your admin user on the merchant store
2. Make sure the email address you use for checking out with your admin user is registered in WooPay, and that WooPay is enabled in the store
3. Add a product to your cart

For the customize.php page:
1. Go to your WP dashboard
2. Go to the customize page ( yoursite.com/wp-admin/customize.php )
3. While customizing, navigate to the checkout page (one using the shortcode, not blocks)
4. Confirm that, after waiting for a bit, the OTP modal (and redirection if you're authenticated) don't take place

For a page preview:
1. Go to your WP dashboard
2. Go edit the checkout page (one using the shortcode, not blocks)
3. Preview the page on a separate window
4. Confirm that, after waiting for a bit, the OTP modal (and redirection if you're authenticated) don't take place

-------------------


**Post merge**
